### PR TITLE
0.3.0: Remove rand from public API, remove from_crypto_rng, update dependencies to the latest.

### PR DESCRIPTION
**Pull request by fadeevab**
Remove `rand` types from `Cocoon` public API
---------------------------------------------------

**Problem**: `cocoon` exposes `RngCore`, `CryptoRng`, `StdRng`, and `ThreadRng` in its public API. Using the incompatible versions of the crate breaks the build.

❌ `Cocoon::from_crypto_rng` removed from public API, because it cannot be implemented without generic type and dependency on `rand` in `Cocoon` struct.

`enum RngVariant` is a wrapper around of `StdRng` and `ThreadRng` for "nostd" builds (no `alloc` crate, no `Box`).

Bumping versions of dependencies
---------------------------------------

⚠ Test vectors of encryption methods are updated. StdRng was changed from ChaCha20 to ChaCha12 in the new version of `rand`, which leads to a change in seed generation. It's okay since decryption works in a backward-compatible manner as expected. Also, it is actually expected to have a random seed in a wild, so tests just unveiled the issue which is good.

### CODE REVIEW

**Comment by fadeevab** [comment]
@bhgomes Please, review. :)

**Comment by bhgomes** [comment]
Thanks for the PR! I'll take a look at it tonight.

**Review by bhgomes — CHANGES_REQUESTED** [review]
LGTM. I would just run `clippy` again and fix those redundant references and the comments I made above.

File: src/lib.rs [hunk 1]
```diff
@@ -284,6 +284,15 @@ use std::io::{Read, Write};
 use format::FormatPrefix;
 use header::{CocoonConfig, CocoonHeader};
 
+// Enumeration is needed to avoid dynamic allocation (important for "nostd" build).
+#[allow(clippy::clippy::large_enum_variant)]
```
[the remark this answers is not in the document]
Comment (fadeevab) [reply src/lib.rs:288]: Right, for some reason both notions work 🤷‍♂️ I double-checked

[the remark this answers is not in the document]
Comment (fadeevab) [reply src/lib.rs:288]: Also, clippy doesn't show me those redundant references warning

**Comment by bhgomes** [comment]
Oh, sorry, I'm seeing the extra `clippy` warnings cause I'm using nightly `clippy`. Up to you if you want to follow that.

**Comment by fadeevab** [comment]
@bhgomes Hey! Maybe, before I publish 0.3.0, you could try the `master` branch locally?.. Thank you for your help!

**Comment by bhgomes** [comment]
@fadeevab It looks good to me. Thanks for the upgrade!

**Comment by fadeevab** [comment]
@bhgomes Published. Thank you!

**Pull request merged**